src/snowflake/snowpark/dataframe.py

```diff
@@ -1072,8 +1073,8 @@ def select(
         if self._select_statement:
             if join_plan:
                 return self._with_plan(
-                    SelectStatement(
-                        from_=SelectSnowflakePlan(
+                    self._session._analyzer.create_select_statement(
+                        from_=self._session._analyzer.create_select_snowflake_plan(
```

Comment:
this is our efforts to avoid `if isinstance(session._conn, MockServerConnection)` everywhere by introducing new interface (I think in Java it's called "polymorphism"?)

create_select_statement will create `MockSelectStatement` is it's a mock connection and `SelectStatement` when it's a lieve connection